### PR TITLE
boards/sam3-based: Model features in Kconfig

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
This models the features provided by the boards based on the `sam3` family of CPUs:
- `arduino-due`
- `udoo`

### Testing procedure
- Check symbol organization and naming
- Green CI: `tests/kconfig_features` should pass for both boards

### Issues/PRs references
Part of #14148

### CODE REVIEW

File: tests/kconfig_features/Makefile [hunk 1]
```diff
@@ -1,6 +1,7 @@
 include ../Makefile.tests_common
 
-BOARD_WHITELIST += arduino-duemilanove \
+BOARD_WHITELIST += arduino-due \
+                   arduino-duemilanove \
```
Comment (aabadie) [inline tests/kconfig_features/Makefile:4]: This board is not based on sam3 but atmega328p.
```suggestion
```

Comment (aabadie) [reply tests/kconfig_features/Makefile:4]: Sorry, it was already there....

**Comment by aabadie** [comment]
Also needs a rebase

**Comment by leandrolanzieri** [comment]
Rebased

**Pull request merged**